aiohttp/client.py

```diff
@@ -566,7 +566,13 @@ async def _request(
                         elif not scheme:
                             parsed_url = url.join(parsed_url)
 
-                        if url.origin() != parsed_url.origin():
+                        https_redirect = (
```

Comment:
It's probably better to be verbose
```suggestion
                        is_same_host_https_redirect = (
```